### PR TITLE
chore: adapt to the new API for NamedTopology builders and runtimes

### PR DESCRIPTION
**Pull request by ableegoldman**
Extracting some self-contained pieces of the mega-PR [#8219](https://github.com/confluentinc/ksql/pull/8219) before I tackle any of the large refactoring that was suggested.

This PR just prepares only the necessary changes for us to finally merge [https://github.com/apache/kafka/pull/11272](#11272)  without breaking the build. Since we were already starting up an empty KafkaStreams, the only changes we need here are to get the `TopologyBuilder` from this KafkaStreams

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryPlan.java
Comment (ableegoldman) [inline ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryPlan.java:88]: just a small fix on the side that I kept in here

File: ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java
Comment (ableegoldman) [inline ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java:69]: again, just a small refactoring that I figured we may as well pull into this PR since it's so small

**Comment by ableegoldman** [comment]
@wcarlson5 @rodesai I'm working on breaking up that one large PR into smaller chunks that we can start to merge right away, and unblock some other work, please take a look

**Review by wcarlson5 — APPROVED** [review]
LGTM

**Comment by wcarlson5** [comment]
```09:50:43  [ERROR] COMPILATION ERROR : 
09:50:43  [INFO] -------------------------------------------------------------
09:50:43  [ERROR] /home/jenkins/workspace/entinc_Contributors_ksql_PR-8331/ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java:[95,66] cannot find symbol```

**Comment by ableegoldman** [comment]
@wcarlson5 this isn't going to compile/pass the PR build until [this](https://github.com/apache/kafka/commit/908a6d2ad7807642b917231769c8d3b6a0c1ad16) gets synced over to ccs-kafka, ,and the sync job is currently blocked on conflicts due to the 3.2 version bump. 

I ran the tests locally with the above commit and verified they pass so we should be good to merge this right away after the sync job goes through

**Pull request merged**
